pkg/mountutil/mountutil.go

```diff
@@ -161,5 +158,150 @@ func ProcessFlagV(s string, volStore volumestore.VolumeStore, createDir bool) (*
 		}
 		res.Mount.Options = strutil.DedupeStrSlice(append(res.Mount.Options, unpriv...))
 	}
-	return &res, nil
+
+	return res, nil
+}
+
+func handleVolumeToMount(source string, volStore volumestore.VolumeStore, createDir bool) (volumeSpec, error) {
+	var res volumeSpec
+
+	res.Type = Bind
+	res.Source = source
+
+	// Handle named volumes
+	if isNamedVolume(source) {
+		return handleNamedVolumes(source, volStore)
+	}
+
+	// Handle named pipes
+	if isNamedPipe(source) {
+		res.Type = Npipe
+		return res, nil
+	}
+
+	// Handle relative paths
+	if !filepath.IsAbs(source) {
+		log.L.Warnf("expected an absolute path, got a relative path %q (allowed for nerdctl, but disallowed for Docker, so unrecommended)", source)
+		absPath, err := filepath.Abs(source)
+		if err != nil {
+			return res, fmt.Errorf("failed to get the absolute path of %q: %w", source, err)
+		}
+		res.Source = absPath
+	}
+
+	// Create dir if it does not exist
+	if err := createDirOnHostIfNotExist(source, createDir); err != nil {
+		return res, err
+	}
+
+	return res, nil
+}
+
+func handleAnonymousVolumes(source string, volStore volumestore.VolumeStore) (volumeSpec, error) {
+	if isNamedPipe(source) {
+		return volumeSpec{Type: Npipe, Source: source}, nil
+	}
+
+	var res volumeSpec
+	res.AnonymousVolume = idgen.GenerateID()
+
+	log.L.Debugf("creating anonymous volume %q, for %q", res.AnonymousVolume, source)
+	anonVol, err := volStore.Create(res.AnonymousVolume, []string{})
+	if err != nil {
+		return res, fmt.Errorf("failed to create an anonymous volume %q: %w", res.AnonymousVolume, err)
+	}
+
+	res.Type = Volume
+	res.Source = anonVol.Mountpoint
+	return res, nil
+}
+
+func handleNamedVolumes(source string, volStore volumestore.VolumeStore) (volumeSpec, error) {
+	var res volumeSpec
+	res.Name = source
+	vol, err := volStore.Get(res.Name, false)
+	if err != nil {
+		if errors.Is(err, errdefs.ErrNotFound) {
+			vol, err = volStore.Create(res.Name, nil)
+			if err != nil {
+				return res, fmt.Errorf("failed to create volume %q: %w", res.Name, err)
+			}
+		} else {
+			return res, fmt.Errorf("failed to get volume %q: %w", res.Name, err)
+		}
+	}
+	// src is now an absolute path
+	res.Type = Volume
+	res.Source = vol.Mountpoint
+
+	return res, nil
+}
+
+func getDestination(split []string) (string, error) {
+	var dst string
+
+	if len(split) == 1 {
+		dst = split[0]
+
+		// For anonymous named volumes, the dst value is set to the created volume for the source
+		if isNamedVolume(dst) {
+			return dst, nil
```

Comment:
nit: from this logic ,once `len(split)` is `1`, doesn't matter if `isNamedVolume(dst)`, it will always return `dst, nil` either at L248 or L259, unless L255 happens. Any way of re-arranging this logic; perhaps starting off with error catching L255, etc?